core/src/main/java/org/apache/james/core/MailAddress.java

```diff
@@ -385,6 +385,9 @@ public final boolean equals(Object obj) {
             if (isNullSender() && addr.isNullSender()) {
```

Comment:
You didn't get what I was expecting.

The idea is that it will be simple if both _if_ statements return the same value `return true` or `return false`

Reply:
Another solution might be to introduce methods for both tests: `areBothSendersNull` & `isOnlyOneSenderNull`